inc/Addon/Varnish/Subscriber.php

```diff
@@ -116,4 +117,36 @@ public function clean_home( $root, $lang ) {
 		$this->varnish->purge( $home_url );
 		$this->varnish->purge( $home_pagination_url );
 	}
+
+	/**
+	 * Clears URLs in Varnish cache.
+	 *
+	 * @param array $urls The path of home cache file.
+	 */
+	public function clean_urls( $urls ) {
+		if ( ! $this->should_purge() ) {
+			return;
+		}
+
+		foreach ( $urls as $data ) {
+			foreach ( $data['files'] as $file_path ) {
+				if ( strpos( $file_path, '#' ) ) {
+					// URL with query string.
+					continue;
+				} else {
+					$file_path         = untrailingslashit( $file_path );
+					$data['home_path'] = untrailingslashit( $data['home_path'] );
+					$data['home_url']  = untrailingslashit( $data['home_url'] );
+					if ( '/' === substr( get_option( 'permalink_structure' ), -1 ) ) {
+						$file_path         .= '/';
+						$data['home_path'] .= '/';
+						$data['home_url']  .= '/';
+					}
+				}
+
+				$url = str_replace( $data['home_path'], $data['home_url'], $file_path );
```

Comment:
Do we have a function that handles this path to url conversion? If yes, can we reuse it here?

Reply:
we have one for URL -> path, but not for path -> URL yet

Reply:
Okay. We will have both of these for reuse in the new Filesystem module. I'll add this code to my running list where we can simplify once that module is ready. Thanks.